pyuvdata/uvdata/uvdata.py

```diff
@@ -1412,580 +1659,489 @@ def reorder_blts(self, order='time', minor_order=None, conj_convention=None, uvw
+                                               frame='itrs', obstime=obs_time)
+            frame_telescope_location = itrs_telescope_location.transform_to(phase_frame)
+            itrs_lat_lon_alt = self.telescope_location_lat_lon_alt
+
+            if use_ant_pos:
+                ant_uvw = uvutils.phase_uvw(self.telescope_location_lat_lon_alt[1],
+                                            self.telescope_location_lat_lon_alt[0],
+                                            self.antenna_positions)
+
+                for bl_ind in inds:
+                    ant1_index = np.where(self.antenna_numbers == self.ant_1_array[bl_ind])[0][0]
+                    ant2_index = np.where(self.antenna_numbers == self.ant_2_array[bl_ind])[0][0]
+                    self.uvw_array[bl_ind, :] = ant_uvw[ant2_index, :] - ant_uvw[ant1_index, :]
+
+            else:
+                uvws_use = self.uvw_array[inds, :]
+
+                uvw_rel_positions = uvutils.unphase_uvw(frame_phase_center.ra.rad,
+                                                        frame_phase_center.dec.rad,
+                                                        uvws_use)
+
+                # astropy 2 vs 3 use a different keyword name
+                rep_keyword = 'representation_type'
+                setattr(frame_telescope_location, rep_keyword, 'cartesian')
+
+                rep_dict = {}
+                rep_dict[rep_keyword] = 'cartesian'
+                frame_uvw_coord = SkyCoord(x=uvw_rel_positions[:, 0] * units.m + frame_telescope_location.x,
+                                           y=uvw_rel_positions[:, 1] * units.m + frame_telescope_location.y,
+                                           z=uvw_rel_positions[:, 2] * units.m + frame_telescope_location.z,
+                                           frame=phase_frame, obstime=obs_time,
+                                           **rep_dict)
+
+                itrs_uvw_coord = frame_uvw_coord.transform_to('itrs')
+
+                # now convert them to ENU, which is the space uvws are in
+                self.uvw_array[inds, :] = uvutils.ENU_from_ECEF(itrs_uvw_coord.cartesian.get_xyz().value.T,
+                                                                *itrs_lat_lon_alt)
+
+        # remove phase center
+        self.phase_center_frame = None
+        self.phase_center_ra = None
+        self.phase_center_dec = None
+        self.phase_center_epoch = None
+        self.set_drift()
+
+    def phase(self, ra, dec, epoch='J2000', phase_frame='icrs',
+              use_ant_pos=False, allow_rephase=True, orig_phase_frame=None):
         """
-        Combine two UVData objects along frequency, polarization and/or baseline-time.
+        Phase a drift scan dataset to a single ra/dec at a particular epoch.
+
+        See the phasing memo under docs/references for more documentation.
+
+        Tested against MWA_Tools/CONV2UVFITS/convutils.
+        Will not phase already phased data.
```

Comment:
This is not necessarily true anymore since we can unphase and rephase in one call right?

Reply:
You're right, good catch!